sim/common/wotlk/enchant_effects.go

```diff
@@ -197,6 +195,22 @@ func init() {
 				}
 			},
 		})
+
+		core.RegisterOnItemSwap(3789, func(sim *core.Simulation) {
+			mh = character.Equip[proto.ItemSlot_ItemSlotMainHand].Enchant.EffectID == 3789
+			oh = character.Equip[proto.ItemSlot_ItemSlotOffHand].Enchant.EffectID == 3789
+
+			procMask = core.GetMeleeProcMaskForHands(mh, oh)
+			ppmm = character.AutoAttacks.NewPPMManager(1.0, procMask)
+			if ppmm.Chance(procMask) == 0 {
+				aura.Deactivate(sim)
+				return
+			}
+
+			if !aura.IsActive() {
```

Comment:
This check is not necessary, `Active` already handles the case where the aura isn't active.